src/math/PercentageMath.sol

```diff
@@ -99,26 +99,27 @@ library PercentageMath {
         }
     }
 
-    /// @notice Executes a weighted average (x * (1 - p) + y * p), rounded up.
+    /// @notice Computes the bps-based weighted average (x * (1 - p) + y * p), rounded half up.
     /// @param x The first value, with a weight of 1 - percentage.
     /// @param y The second value, with a weight of percentage.
-    /// @param percentage The weight of y, and complement of the weight of x.
-    /// @return z The result of the weighted average.
+    /// @param percentage The weight of y, and complement of the weight of x (in bps).
+    /// @return z The result of the bps-based weighted average.
     function weightedAvg(
         uint256 x,
         uint256 y,
         uint256 percentage
     ) internal pure returns (uint256 z) {
-        // Must revert if
+        // Revert if
         //     percentage > PERCENTAGE_FACTOR
-        // or if
+        // Overflow if
         //     y * percentage + HALF_PERCENTAGE_FACTOR > type(uint256).max
-        //     <=> percentage > 0 and y > (type(uint256).max - HALF_PERCENTAGE_FACTOR) / percentage
-        // or if
+        // <=> percentage > 0 and y > (type(uint256).max - HALF_PERCENTAGE_FACTOR) / percentage
```

Comment:
the <=> no longer works here

Reply:
But if `percentage = 0` then `HALF_PERCENTAGE_FACTOR > type(uint256).max` which is a contradiction, hence the equivalence?

Reply:
But if `percentage = 0` then `HALF_PERCENTAGE_FACTOR > type(uint256).max` which is a contradiction
Why isn't it equivalent?

Reply:
ok ok I get it. 

I read:

revert if A and overflow if B <=> C and D (which makes no sense)

whereas I should read:

revert if A and overflow if (B <=> C)